src/experiments/title-generation/components/TitleToolbarWrapper.tsx

```diff
@@ -0,0 +1,364 @@
+ *
+ * This component uses DOM manipulation to attach the toolbar to the title field
+ * in normal editing mode (non-template mode).
+ */
+
+/**
+ * External Dependencies.
+ */
+import React from 'react';
+import { createRoot, useEffect } from '@wordpress/element';
+
+/**
+ * Internal Dependencies.
+ */
+import TitleToolbar from './TitleToolbar';
+
+/**
+ * TitleToolbarWrapper component.
+ *
+ * Attaches the toolbar to the title field in normal editing mode.
+ *
+ * @return {JSX.Element} The wrapper component.
+ */
+function TitleToolbarWrapper(): JSX.Element {
+	useEffect(() => {
+		let isAttached = false;
+		let root: ReturnType<typeof createRoot> | null = null;
+		let removeTitleListeners: (() => void) | null = null;
+		let removeToolbarListeners: (() => void) | null = null;
+		let observer: MutationObserver | null = null;
+		let titleInput: HTMLElement | null = null;
+		let toolbarContainer: HTMLElement | null = null;
+		let wrapperContainer: HTMLElement | null = null;
+
+		// Find the editor iframe
+		const getEditorDocument = (): Document | null => {
+			// Try to find the iframe that contains the editor
+			const iframe = document.querySelector(
+				'iframe[name="editor-canvas"], iframe.wp-block-editor-iframe__iframe'
+			) as HTMLIFrameElement | null;
+
+			if (iframe && iframe.contentDocument) {
+				return iframe.contentDocument;
+			}
+
+			return null;
+		};
+
+		// Create a reusable retry function
+		const createRetry = (
+			checkFn: () => boolean,
+			callback: () => void,
+			maxRetries: number,
+			delay: number = 200
+		): (() => void) => {
+			let retryCount = 0;
+			let timeoutId: NodeJS.Timeout | null = null;
+
+			const retry = () => {
+				if (retryCount < maxRetries && !isAttached) {
+					retryCount++;
+					timeoutId = setTimeout(() => {
+						timeoutId = null;
+						if (checkFn()) {
+							callback();
+						} else {
+							retry();
+						}
+					}, delay);
+				}
+			};
+
+			// Store timeout ID for cleanup
+			const start = () => {
+				if (!timeoutId) {
+					retry();
+				}
+			};
+
+			return start;
+		};
+
+		// Create reusable focus/blur handlers
+		const createFocusBlurHandlers = (
+			onFocus: () => void,
+			onBlur: () => void,
+			blurDelay: number = 10
+		) => {
+			return {
+				focus: onFocus,
+				blur: () => {
+					setTimeout(onBlur, blurDelay);
+				},
+			};
+		};
+
+		// Setup event listeners on an element
+		const setupEventListeners = (
+			element: HTMLElement,
+			handlers: { focus: () => void; blur: () => void },
+			useFocusIn: boolean = false
+		) => {
+			const focusEvent = useFocusIn ? 'focusin' : 'focus';
+			const blurEvent = useFocusIn ? 'focusout' : 'blur';
+
+			element.addEventListener(focusEvent, handlers.focus);
+			element.addEventListener(blurEvent, handlers.blur);
+
+			return () => {
+				element.removeEventListener(focusEvent, handlers.focus);
+				element.removeEventListener(blurEvent, handlers.blur);
+			};
+		};
```

Comment:
This can be pulled out, too.

Reply:
Fixed in https://github.com/WordPress/ai/pull/83/commits/378116f187322a3262df4da0ed4ef72b53d1abbe